std/encoding.d

```diff
@@ -2250,24 +2138,27 @@ body
             immutable minReservePlace = 6;
         }
 
-        Dst[] buffer = new Dst[s.length];
```

Comment:
In order to do that, I would need to make half of std.encoding pure.

I can restore the old code as @trusted as an interim step, or I can make everything pure and we can come back to this in a week or two.